common/lib/xmodule/xmodule/capa_base.py

```diff
@@ -1233,7 +1233,12 @@ def submit_problem(self, data, override_time=False):
             # without a stack trace
             else:
                 # Translators: {msg} will be replaced with a problem's error message.
-                msg = inst.message
+                try:
+                    # only return the error value of the exception
+                    message = inst.message.split("\\n")[-2].split(": ", 1)[1]
+                    msg = cgi.escape(message)
```

Comment:
I take your point about missing the escaping with the IndexError, so I moved it outside the try block. 

But on line 1230, we're escaping the a different string, the full traceback.